src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -1,16 +1,122 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.Mining;
+using Stratis.Bitcoin.Utilities;
 
 namespace Stratis.Bitcoin.Features.PoA
 {
-    public class PoAMiner
+    // TODO POA create an interface like we did with PoSMinting
+    public class PoAMiner : IDisposable
     {
-        // TODO POA. First add creation of block template and later apply signatures and other stuff. We need bare minimum to test syncing and payloads
+        // TODO POA. Implement miner properly later. Right now we need bare minimum to test syncing and payloads.
 
-        public void StartMining()
+        private const bool MineDuringIBD = true; // TODO POA just for tests
+
+        private readonly IConsensusManager consensusManager;
+
+        private readonly IDateTimeProvider dateTimeProvider;
+
+        private readonly ILogger logger;
+
+        private readonly Network network;
+
+        /// <summary>
+        /// A cancellation token source that can cancel the mining processes and is linked to the <see cref="INodeLifetime.ApplicationStopping"/>.
+        /// </summary>
+        private CancellationTokenSource cancellation;
+
+        private readonly IInitialBlockDownloadState ibdState;
+
+        private readonly PoABlockDefinition blockDefinition;
+
+        private Task miningTask;
+
+        public PoAMiner(
+            IConsensusManager consensusManager,
+            IDateTimeProvider dateTimeProvider,
+            Network network,
+            INodeLifetime nodeLifetime,
+            ILoggerFactory loggerFactory,
+            IInitialBlockDownloadState ibdState,
+            PoABlockDefinition blockDefinition)
+        {
+            this.consensusManager = consensusManager;
+            this.dateTimeProvider = dateTimeProvider;
+            this.network = network;
+            this.ibdState = ibdState;
+            this.blockDefinition = blockDefinition;
+
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.cancellation = CancellationTokenSource.CreateLinkedTokenSource(new[] { nodeLifetime.ApplicationStopping });
```

Comment:
Why create a linked cancellation, just wondering.

Reply:
copied from PoW miner. It's for having an ability to stop miner manually or via lifetime